crates/signed_note/src/lib.rs

```diff
@@ -248,6 +248,10 @@ pub trait Verifier {
 
     /// Reports whether sig is a valid signature of msg.
     fn verify(&self, msg: &[u8], sig: &[u8]) -> bool;
+
+    /// Extracts a Unix timestamp in milliseconds from the given signature bytes, if defined. Errors if
+    /// the signature is malformed.
+    fn extract_timestamp_millis(&self, sig: &[u8]) -> Result<Option<u64>, ()>;
```

Comment:
Nice addition to the trait! If we define the `UnixTimestamp` alias in some shared crate later we can use it here.
```suggestion
    fn extract_timestamp_millis(&self, sig: &[u8]) -> Result<Option<UnixTimestamp>, ()>;
```